Add odk:obsolete-replace command

matentzn opened the pull request:
Adding the odk:obsolete-replace command to replace a key command of the owltools stack, eg:

```
owltools --use-catalog edit.obo --obsolete-replace http://purl.obolibrary.org/obo/VBO_0000012 http://purl.obolibrary.org/obo/VBO_0000038 -o -f obo edit.obo
```

The command essentially:

1. obsoletes one term
2. moves selected properties over to the replacement term

I changed the interface a little but (for now) left the (owltools) implementation more or less what it was. I tested this in Mondo and it works as expected.

I contemplated adding this to my custom Monarch plugin but since this is used by at least two ontologies I know off, I thought ODK ROBOT plugin might be the right place. Correct me if I am wrong, happy to migrate this over.

matentzn committed:
Add odk:obsolete-replace command

Adding the odk:obsolete-replace command to replace a key command of the owltools stack, eg:

```
owltools --use-catalog edit.obo --obsolete-replace http://purl.obolibrary.org/obo/VBO_0000012 http://purl.obolibrary.org/obo/VBO_0000038 -o -f obo edit.obo
```

The command essentially:

1. obsoletes one term
2. moves selected properties over to the replacement term

I changed the interface a little but (for now) left the (owltools) implementation more or less what it was. I tested this in Mondo and it works as expected.

gouttegd commented on src/main/java/org/incenp/obofoundry/odk/EntityReplacementVisitor.java:
```diff
@@ -0,0 +1,89 @@
+/*
+ * ODK ROBOT Plugin
+ * Copyright © 2025 Damien Goutte-Gattat
```
I suppose you copied that over from any of the existing files in the repo, but that’s not appropriate: I did not write a single line of this particular file, so I have no business being the copyright owner here.

Is the code in this file a _direct_ copy of a piece of code in OWLTools? In that case the ownership should be the same as the current ownership in OWLTools.

If it is _your_ code (even if strongly inspired by the OWLTools code), then _you_ are the copyright owner here.

matentzn replied:
Its "strongly inspired" - I added myself, and a note to the source of inspiration.

gouttegd replied:
Thanks. You know I am a stickler for this kind of things. :)

gouttegd commented on src/main/java/org/incenp/obofoundry/odk/ObsoleteReplaceCommand.java:
```diff
@@ -0,0 +1,195 @@
+/*
+ * ODK ROBOT Plugin
+ * Copyright © 2025 Damien Goutte-Gattat
```
Same remark as above.

gouttegd commented on README.md:
```diff
@@ -14,8 +14,10 @@ The following commands are provided by the plugin:
-* `odk:import`: to add/remove import declarations.
+* `odk:normalize`: to "normalise" an ontology;
+* `odk:import`: to add/remove import declarations;
+* `odk:obsolete-replace`: to obsolete an entity and replace all its usages
```
It’d be nice (though I will not _demand_ that before I accept the PR) to have even a basic documentation page for this command, like we have for the other commands (e.g. [odk:check-align](https://incatools.github.io/odk-robot-plugin/alignment.html)). The sources for the documentation are there: https://github.com/INCATools/odk-robot-plugin/tree/main/src/site/markdown.

matentzn replied:
DId

gouttegd reviewed (changes requested):
No objection to adding such a command to the ODK plugin. Seems to me that it is perfectly “in scope”.

However I must ask you to please fix the copyright notices.

matentzn committed:
Update copyright holder to Nico Matentzoglu

Changed the copyright notice in EntityReplacementVisitor.java and ObsoleteReplaceCommand.java from Damien Goutte-Gattat to Nico Matentzoglu.

matentzn committed:
Add documentation for odk:obsolete-replace command

Introduces a markdown file describing the usage, options, and workflow for the odk:obsolete-replace command.

matentzn committed:
Add attribution comment to ObsoleteReplaceCommand

Added a comment noting that the command was inspired by the OWLTools CommandRunner implementation. This provides proper attribution and context for future maintainers.

gouttegd commented:
Do you need a 0.3.0 release with that new command in?

matentzn commented:
Eventually - but if you have things in the pipe you want to implement here before the next minor ODK release it can wait. What I am saying - I need this with the next minor ODK release :) Until then it can wait. And its of the important but not urgent variety of tasks.

gouttegd commented:
I currently have no immediate (or even not-so-immediate) plans for the ODK plugin, so happy to release version 0.3.0 soon.

Then the question becomes, when do we want to release ODK 1.6.1. Moving the discussion to the ODK repository…
